Add testcases to test renaming directories as well as files. Remove the
l bit from the ok perms as that's no longer needed.
---
 tests/regression/subdomain/rename.sh |   60 +++++++++++++++++++----------------
 1 file changed, 34 insertions(+), 26 deletions(-)

Index: subdomain/rename.sh
===================================================================
--- subdomain.orig/rename.sh
+++ subdomain/rename.sh
@@ -9,9 +9,9 @@
 #	License.
 
 #=NAME rename
-#=DESCRIPTION 
-# The rename system call changes the name of a file in the filesystem.  The 
-# test verifies that this operation (which involves AppArmor write and link 
+#=DESCRIPTION
+# The rename system call changes the name of a file in the filesystem.
+# The test verifies that this operation (which involves AppArmor write
 # permission checks) functions correctly for a confined process.
 #=END
 
@@ -24,63 +24,71 @@ bin=$pwd
 
 file1=$tmpdir/file1
 file2=$tmpdir/file2
+dir1=$tmpdir/dir1
+dir2=$tmpdir/dir2
 
-okfile1perm=rwl
+okfile1perm=rw
 badfile1perm1=r
 badfile1perm2=w
 okfile2perm=w
 badfile2perm=r
 
-# PASS TEST
+reset_test() {
+	touch $file1
+	chmod 600 $file1
+	test -d $dir1 || mkdir $dir1
+	chmod 700 $dir1
+}
 
-touch $file1 
-chmod 600 $file1
+# PASS TEST
+reset_test
 
-genprofile $file1:$okfile1perm $file2:$okfile2perm
+genprofile $file1:$okfile1perm $file2:$okfile2perm $dir1:$okfile1perm $dir2:$okfile2perm
 
-runchecktest "RENAME RWL W" pass $file1 $file2
+runchecktest "RENAME RW W" pass $file1 $file2
+runchecktest "RENAME RW W (dir)" pass $dir1 $dir2
 
 # FAILURE TEST (1) - Bad permissions on target
 
-touch $file1 
-chmod 600 $file1
+reset_test
 
-genprofile $file1:$okfile1perm $file2:$badfile2perm
+genprofile $file1:$okfile1perm $file2:$badfile2perm $dir1:$okfile1perm $dir2:$badfile2perm
 
-runchecktest "RENAME RWL R" fail $file1 $file2
+runchecktest "RENAME RW R" fail $file1 $file2
+runchecktest "RENAME RW R (dir)" fail $dir1 $dir2
 
 # FAILURE TEST (2) - no permissions on target
 
-touch $file1 
-chmod 600 $file1
+reset_test
 
-genprofile $file1:$okfile1perm
+genprofile $file1:$okfile1perm $dir1:$okfile1perm
 
-runchecktest "RENAME RWL -" fail $file1 $file2
+runchecktest "RENAME RW -" fail $file1 $file2
+runchecktest "RENAME RW - (dir)" fail $dir1 $dir2
 
 # FAILURE TEST (3) - Bad permissions on source
 
-touch $file1 
-chmod 600 $file1
+reset_test
 
-genprofile $file1:$badfile1perm1 $file2:$okfile2perm
+genprofile $file1:$badfile1perm1 $file2:$okfile2perm $dir1:$badfile1perm1 $dir2:$okfile2perm
 
 runchecktest "RENAME R W" fail $file1 $file2
+runchecktest "RENAME R W (dir)" fail $dir1 $dir2
 
 # FAILURE TEST (4) - Bad permissions on source
 
-touch $file1 
-chmod 600 $file1
+reset_test
 
-genprofile $file1:$badfile1perm2 $file2:$okfile2perm
+genprofile $file1:$badfile1perm2 $file2:$okfile2perm $dir1:$badfile1perm2 $dir2:$okfile2perm
 
 runchecktest "RENAME W W" fail $file1 $file2
+runchecktest "RENAME W W (dir)" fail $dir1 $dir2
 
 # FAILURE TEST (5) - No permissions on source
 
-touch $file1 
-chmod 600 $file1
+reset_test
 
-genprofile $file2:$okfile2perm
+genprofile $file2:$okfile2perm $dir2:$okfile2perm
 
 runchecktest "RENAME - W" fail $file1 $file2
+runchecktest "RENAME - W (dir)" fail $dir1 $dir2
